respx/router.py

```diff
@@ -253,11 +253,12 @@ def resolve(self, request: httpx.Request) -> Optional[httpx.Response]:
                 # Pass-through request
                 response = None
 
-            else:
-                # Mocked response
-                assert isinstance(mock, httpx.Response)
+            elif isinstance(mock, httpx.Response):
                 response = mock
 
+            else:
+                raise TypeError(f"RESPX: {mock!r} is not a instance of httpx.Response")
```

Comment:
😉 
```suggestion
                raise TypeError(f"RESPX: {mock!r} is not an instance of httpx.Response")
```

Reply:
btw, this line will never be reached if we type check in the `Route.return_value` setter as I mention 😉